### ISSUE TITLE
Browser crashes on iOS with too many layers

### ISSUE THREAD

**Issue by jasontk19** [openlayers/openlayers#14171]
**Describe the bug**
We have noticed that there is a bug that is crashing our app when there are too many layers added that was introduced somewhere in the 6.14.2 release of OpenLayers.  In the CodeSandBox examples below you can see that, in the working version, there is no problem loading 20 different layers onto the map when viewing from an iOS device (either Chrome or Safari).  Two dev versions later, you can see that this consistently crashes the browser on page load.  Oddly enough, the version in between these two seemed to _sometimes_ crash and sometimes it did not, though that could have been a timing/browser caching issue on my end.

The crash looks like this in the browser:
![File (1)](https://user-images.githubusercontent.com/1480000/193897183-cea58194-ffbf-4870-b65b-6e2cc9978837.jpg)

In trying to debug this, I also gathered device error logs for both browsers (which produced similar error messages):
Safari:
![Screen Shot 2022-10-04 at 8 30 31 AM](https://user-images.githubusercontent.com/1480000/193897016-be7419c9-ce73-4a11-9fbc-e0c4194e68f4.png)

Chrome:
![Screen Shot 2022-10-04 at 8 28 12 AM](https://user-images.githubusercontent.com/1480000/193897020-7da20d9a-376c-4ffa-9a99-be29a77035a8.png)

**To Reproduce**
You can see the issue occurring in this CodeSandBox:
https://codesandbox.io/s/ios-crash-broken-6-14-2-dev-1657990973093-ivfu85
Direct Link for viewing on mobile: https://ivfu85.csb.app/

And, in the previous version you can see it is not an issue in this CodeSandBox:
https://codesandbox.io/s/ios-crash-working-6-14-2-dev-1657910026604-65so4n
Direct link for viewing on mobile: https://65so4n.csb.app/

**Expected behavior**
The upper limit for number of layers that can be rendered without crashing should be higher (and was in previous version)

**Comment by tschaub** [comment]
I'm curious why you are using the `className` in the layer options.  Removing that makes your example work for me without crashing (https://peqf8e.csb.app/).

**Comment by mike-000** [comment]
Yes, the problem is caused by using `className`.  #13823 bases layer canvas size on device pixel ratio which makes it more likely the canvas can be reused by other layers, but that can make it bigger, so `className` which prevents reuse should only be used if essential which, with the removal of `forEachLayerAtPixel`, is now unlikely.  Even the grayscale background in the Semi-Transparent Layer example could be easily done using a `postrender` event instead of a `className` https://codesandbox.io/s/semi-transparent-layer-forked-d2p398?file=/main.js

**Comment by jasontk19** [comment]
I am not sure that we even need to use `className` in our app, it's likely that we do not.

Thank you for identifying the presence of this option as the problem.  I will investigate more to see if we can remove the `className` without any side effects on our end.

**Comment by jasontk19** [comment]
That's good to know, unfortunately we have both raster and vector based imagery and so it sounds like we wouldn't be able to take advantage of this.
